src/mbgl/renderer/sources/render_raster_dem_source.cpp

```diff
@@ -82,51 +82,61 @@ static void fillBorder(const RasterDEMTile& tile, const RasterDEMTile& borderTil
 
     if (tileDEM->isLoaded() && borderDEM->isLoaded()){
         tileDEM->backfillBorder(*borderDEM, dx, dy);
+        // update the bitmask to indicate that this tiles have been backfilled by flipping the relevant bit
+        tile.neighboringTiles = tile.neighboringTiles | mask;
+        std::cout << "backfilled: " << int(tile.id.canonical.z)<< "/" << tile.id.canonical.x << "/" << tile.id.canonical.y << "\n";
+        std::cout << "with neighbor " << int(mask) << " : " << int(borderTile.id.canonical.z)<< "/" << borderTile.id.canonical.x << "/" << borderTile.id.canonical.y << "\n";
+        tileBucket->prepared = false;
     }
-    tileBucket->prepared = false;
-    borderBucket->prepared = false;
 }
 
 void RenderRasterDEMSource::onTileChanged(Tile& tile){
     RasterDEMTile& demtile = static_cast<RasterDEMTile&>(tile);
 
+    std::map<DEMTileNeighbors, DEMTileNeighbors> opposites = {
+        { DEMTileNeighbors::Left, DEMTileNeighbors::Right },
+        { DEMTileNeighbors::Right, DEMTileNeighbors::Left },
+        { DEMTileNeighbors::TopLeft, DEMTileNeighbors::BottomRight },
+        { DEMTileNeighbors::TopCenter, DEMTileNeighbors::BottomCenter },
+        { DEMTileNeighbors::TopRight, DEMTileNeighbors::BottomLeft },
+        { DEMTileNeighbors::BottomRight,  DEMTileNeighbors::TopLeft },
+        { DEMTileNeighbors::BottomCenter, DEMTileNeighbors:: TopCenter },
+        { DEMTileNeighbors::BottomLeft, DEMTileNeighbors::TopRight }
+    };
+
     if (tile.isRenderable() && demtile.neighboringTiles != 0b11111111) {
         const CanonicalTileID canonical = tile.id.canonical;
         const uint dim = std::pow(2, canonical.z);
-        const int px = (canonical.x - 1 + dim) % dim;
+        const uint32_t px = (canonical.x - 1 + dim) % dim;
         const int pxw = canonical.x == 0 ? tile.id.wrap - 1 : tile.id.wrap;
-        const int nx = (canonical.x + 1 + dim) % dim;
+        const uint32_t nx = (canonical.x + 1 + dim) % dim;
         const int nxw = (canonical.x + 1 == dim) ? tile.id.wrap + 1 : tile.id.wrap;
+        const int lowerY = (canonical.y + 1) == dim ? canonical.y : canonical.y + 1;
+        const int upperY = canonical.y == 0 ? canonical.y : canonical.y - 1;
 
         std::vector<OverscaledTileID> neighbors = {
+            // left and right neighbor
```

Comment:
Could use a comment here to indicate that the order of the tile ids defined here must be kept in sync with the order of enum values defined in `DEMTileNeighbors.`

Alternatively, you can remove this vector and add a lambda function to dynamically compute the id in each iteration of the for-loop.